code/modules/reagents/chemistry/recipes/others.dm

```diff
@@ -860,3 +860,155 @@
+
+/datum/chemical_reaction/ice_cream_lemon_sorbet
+	name = /datum/reagent/consumable/ice_cream/lemon_sorbet
+	id = /datum/reagent/consumable/ice_cream/lemon_sorbet
+	results = list(/datum/reagent/consumable/ice_cream/lemon_sorbet = 10)
+	required_reagents = list(/datum/reagent/consumable/ice_cream = 10, /datum/reagent/consumable/lemonjuice = 2)
+
+/datum/chemical_reaction/ice_cream_caramel
+	name = /datum/reagent/consumable/ice_cream/caramel
+	id = /datum/reagent/consumable/ice_cream/caramel
+	results = list(/datum/reagent/consumable/ice_cream/caramel = 10)
+	required_reagents = list(/datum/reagent/consumable/ice_cream = 10, /datum/reagent/consumable/caramel = 2)
+
+/datum/chemical_reaction/ice_cream_banana
+	name = /datum/reagent/consumable/ice_cream/banana
+	id = /datum/reagent/consumable/ice_cream/banana
+	results = list(/datum/reagent/consumable/ice_cream/banana = 10)
+	required_reagents = list(/datum/reagent/consumable/ice_cream = 10, /datum/reagent/consumable/banana = 2)
+
+/datum/chemical_reaction/ice_cream_orange_creamsicle
+	name = /datum/reagent/consumable/ice_cream/orange_creamsicle
+	id = /datum/reagent/consumable/ice_cream/orange_creamsicle
+	results = list(/datum/reagent/consumable/ice_cream/orange_creamsicle = 10)
+	required_reagents = list(/datum/reagent/consumable/ice_cream = 10, /datum/reagent/consumable/orangejuice = 2)
+
+/datum/chemical_reaction/ice_cream_peach
+	name = /datum/reagent/consumable/ice_cream/peach
+	id = /datum/reagent/consumable/ice_cream/peach
+	results = list(/datum/reagent/consumable/ice_cream/peach = 10)
+	required_reagents = list(/datum/reagent/consumable/ice_cream = 10, /datum/reagent/consumable/peachjuice = 2)
+
+/datum/chemical_reaction/ice_cream_cherry_chocolate
+	name = /datum/reagent/consumable/ice_cream/cherry_chocolate
+	id = /datum/reagent/consumable/ice_cream/cherry_chocolate
+	results = list(/datum/reagent/consumable/ice_cream/cherry_chocolate = 10)
+	required_reagents = list(/datum/reagent/consumable/ice_cream/chocolate = 10, /datum/reagent/consumable/cherryjelly = 2)
+
+/datum/chemical_reaction/ice_cream_meat
+	name = /datum/reagent/consumable/ice_cream/meat
+	id = /datum/reagent/consumable/ice_cream/meat
+	results = list(/datum/reagent/consumable/ice_cream/meat = 10)
+	required_reagents = list(/datum/reagent/consumable/ice_cream = 10, /datum/reagent/liquidgibs = 2)
+
+/datum/chemical_reaction/ice_cream_solidification
+	name = "Ice Cream Solidification"
+	id = "ice_cream"
+	required_reagents = list(/datum/reagent/consumable/ice_cream = 10, /datum/reagent/consumable/ice = 2)
+	mob_react = FALSE
+
+	//What scoop it will create
+	var/ice_cream_scoop = /obj/item/reagent_containers/food/snacks/ice_cream_scoop
+
+/datum/chemical_reaction/ice_cream_solidification/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	for(var/i = 1, i <= created_volume, i++)
+		new ice_cream_scoop(location)
```

Comment:
there's gotta be a better way to do this, but i can't think of one off hand, so for now i guess it's fine

Reply:
Yea I'm not sure how either.